tests/e2e/lib.sh

```diff
@@ -554,6 +554,35 @@ wait_for_collectors_to_be_operational() {
         return
     fi
 
+    # Ensure collector DaemonSet state is stable
+    local max_retries=12
+    local num_desired=""
+    local num_updated=""
+    local num_ready=""
+    local tmp_kubectl_json=""
+    while (( max_retries > 0 )); do
+        tmp_kubectl_json=$(kubectl get --namespace "${sensor_namespace}" daemonset/collector --output json)
+
+        num_desired=$(jq '.status.desiredNumberScheduled' --raw-output <<< "${tmp_kubectl_json}")
+        num_updated=$(jq '.status.updatedNumberScheduled' --raw-output <<< "${tmp_kubectl_json}")
+        num_ready=$(jq '.status.numberReady' --raw-output <<< "${tmp_kubectl_json}")
+
+        if [[ "$num_desired" != "" && "$num_updated" != "" && "$num_ready" != "" ]] && (( num_desired == num_updated && num_desired == num_ready )); then
+            echo "DaemonSet collector is stable."
+            break
+        fi
+
+        ((max_retries--))
+        echo "DaemonSet collector is still updating. Will retry in 5sec."
+        sleep 5
+    done
+
+    if (( max_retries <= 0 )); then
+        echo "ERROR: DaemonSet collector is not in a good state. Reached maximum retires."
+        exit 1
+    fi
+
```

Comment:
Let's show what's going on while we wait.

```suggestion
    while true; do
        tmp_kubectl_json=$(kubectl get --namespace "${sensor_namespace}" daemonset/collector --output json)

        num_desired=$(jq '.status.desiredNumberScheduled' --raw-output <<< "${tmp_kubectl_json}")
        num_updated=$(jq '.status.updatedNumberScheduled' --raw-output <<< "${tmp_kubectl_json}")
        num_ready=$(jq '.status.numberReady' --raw-output <<< "${tmp_kubectl_json}")

        if [[ "$num_desired" != "" && "$num_updated" != "" && "$num_ready" != "" ]] && (( num_desired == num_updated && num_desired == num_ready )); then
            echo "DaemonSet collector is stable."
            break
        fi

        echo "DaemonSet collector is still updating: desiredNumberScheduled=${num_desired} updatedNumberScheduled=${num_updated} numberReady=${num_ready}"
        ((max_retries--))
        if (( max_retries <= 0 )); then
            echo "ERROR: DaemonSet collector is not in a good state. Reached maximum retries."
            exit 1
        fi
        echo "Will retry in 5sec."
        sleep 5
    done

```